### PR TITLE
[Kotlin Migration] LoadPronounciationActivity

### PR DESCRIPTION
**Pull request by Akshay0701**

### CODE REVIEW

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
```
/home/runner/work/Anki-Android/Anki-Android/AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/LoadPronounciationActivity.kt:54: Error: Variable name should not use field prefixes. [VariableNamingDetector from com.ichi2.anki.lint]
Fix the issues identified by lint, or add the following to your build script to proceed with errors:
    var mSource: String? = null
...
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/LoadPronounciationActivity.kt [hunk 1]
```diff
@@ -56,153 +50,126 @@ import timber.log.Timber;
...
-    protected void onCreate(Bundle savedInstanceState) {
-        super.onCreate(savedInstanceState);
-        Themes.disableXiaomiForceDarkMode(this);
-
-        if (AdaptionUtil.isUserATestClient()) {
-            finishCancel();
-            return;
+open class LoadPronounciationActivity : Activity(), DialogInterface.OnCancelListener {
+    var mSource: String? = null
+    private var mTranslationAddress: String? = null
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/LoadPronounciationActivity.kt:55]: `KotlinCleanup`: `lateinit`

**Review by david-allison-1 — CHANGES_REQUESTED** [review]
one nit, this could do with a manual test

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/LoadPronounciationActivity.kt [hunk 2]
```diff
@@ -214,252 +182,200 @@ public class LoadPronounciationActivity extends Activity implements OnCancelList
...
-
-        private String mAddress;
-
-
-        @Override
-        protected String doInBackground(Void... params) {
-            return HttpFetcher.downloadFileToSdCard(mAddress, mActivity, "pronunc");
+    @Suppress("deprecation") // #7108: AsyncTask
+    private inner class DownloadFileTask : android.os.AsyncTask<Void?, Void?, String?>() {
+        private var mAddress: String? = null
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/LoadPronounciationActivity.kt:201]: cleanup: lateInit (I think)

**Review by david-allison-1 — CHANGES_REQUESTED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/LoadPronounciationActivity.kt [hunk 3]
```diff
@@ -56,153 +50,127 @@ import timber.log.Timber;
...
-
+    @Suppress("deprecation") // #7108: AsyncTask
+    protected inner class BackgroundPost : android.os.AsyncTask<Void?, Void?, String?>() {
+        @KotlinCleanup("comments rearrange")
+        /**
+         * @return Used to know, which of the posts finished, to differentiate.
+         */
+        /**
+         * @param address Used to set the download address
+         */
```
[the remark this answers is not in the document]
Comment (Akshay0701) [reply AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/LoadPronounciationActivity.kt:168]: in kotlin there is no methods of get and set, so where must I put this comment?

**Comment by Akshay0701** [comment]
Tested this on devices, no crash

**Review by david-allison-1 — APPROVED** [review]
Thanks for your time here, LGTM

**Review by mikehardy — APPROVED** [review]

**Pull request merged**